packages/coin-support-xrp/src/operations/createAccounts/index.ts

```diff
@@ -58,22 +59,39 @@ const getAddressesFromDevice: GetAddressesFromDevice<XrpApp> = async params => {
 
 const createAccountFromAddress: IMakeCreateAccountsObservableParams<XrpApp>['createAccountFromAddress'] =
   async (addressDetails, params) => {
-    const coin = xrpCoinList[params.coinId];
-    const name = `${coin.name} ${addressDetails.index + 1}`;
+    const { address, index, balance, txnCount, derivationPath, schemeName } =
+      addressDetails;
+    const { coinId, walletId } = params;
+
+    const coin = xrpCoinList[coinId];
+    const name = `${coin.name} ${index + 1}`;
+
+    let spendableBalance = new BigNumber(balance);
+    if (!spendableBalance.isZero()) {
```

Comment:
We don't need the if condtion, now that you are using max function

Reply:
to avoid the api call?